pkg/plugins/policies/meshtimeout/plugin/v1alpha1/plugin.go

```diff
@@ -263,48 +271,84 @@ func createInboundClusterName(servicePort uint32, listenerPort uint32) string {
-		conf := rules.Compute(uri, meshCtx.Resources)
-		if conf == nil {
-			conf = &outbound.ResourceRule{Conf: []interface{}{api.Conf{}}}
-		}
+func applyToRealResources(
+	rs []*core_xds.Resource,
+	rules outbound.ResourceRules,
+	reader kri.ResourceReader,
+) error {
+	for _, r := range rs {
+		switch envoyResource := r.Resource.(type) {
+		case *envoy_listener.Listener:
+			switch r.ResourceOrigin.ResourceType {
+			case meshservice_api.MeshServiceType:
+			case meshexternalservice_api.MeshExternalServiceType:
+			case meshmultizoneservice_api.MeshMultiZoneServiceType:
+			default:
+				continue
+			}
 
-		for typ, resources := range resType {
-			switch typ {
-			case envoy_resource.ListenerType:
-				err := configureListeners(resources, conf.Conf[0].(api.Conf))
-				if err != nil {
-					return err
-				}
-			case envoy_resource.ClusterType:
-				err := configureClusters(resources, conf.Conf[0].(api.Conf))
-				if err != nil {
-					return err
-				}
+			listenerConf := rules.Compute(*r.ResourceOrigin, reader)
+			if listenerConf == nil {
+				listenerConf = &outbound.ResourceRule{Conf: []interface{}{api.Conf{}}}
 			}
-		}
-	}
-	return nil
-}
 
-func configureListeners(resources []*core_xds.Resource, conf api.Conf) error {
-	for _, resource := range resources {
-		configurer := plugin_xds.ListenerConfigurer{Conf: conf, Protocol: resource.Protocol}
-		err := configurer.ConfigureListener(resource.Resource.(*envoy_listener.Listener))
-		if err != nil {
-			return err
+			configurer := plugin_xds.ListenerConfigurer{Conf: listenerConf.Conf[0].(api.Conf), Protocol: r.Protocol}
+			if err := configurer.ConfigureListener(envoyResource); err != nil {
+				return err
+			}
+
+			for _, fc := range envoyResource.FilterChains {
+				_ = listeners_v3.UpdateHTTPConnectionManager(fc, func(hcm *envoy_hcm.HttpConnectionManager) error {
+					if err := configureRoutes(hcm.GetRouteConfig(), rules, reader); err != nil {
+						return err
+					}
+					return nil
+				})
+			}
+
+		case *envoy_cluster.Cluster:
+			clusterConf := rules.Compute(*r.ResourceOrigin, reader)
+			if clusterConf == nil {
+				clusterConf = &outbound.ResourceRule{Conf: []interface{}{api.Conf{}}}
+			}
+
+			configurer := plugin_xds.ClusterConfigurerFromConf(clusterConf.Conf[0].(api.Conf), r.Protocol)
+			if err := configurer.Configure(envoyResource); err != nil {
+				return err
+			}
 		}
 	}
 	return nil
 }
 
-func configureClusters(resources []*core_xds.Resource, conf api.Conf) error {
-	for _, resource := range resources {
-		configurer := plugin_xds.ClusterConfigurerFromConf(conf, resource.Protocol)
-		err := configurer.Configure(resource.Resource.(*envoy_cluster.Cluster))
-		if err != nil {
-			return err
+func configureRoutes(
+	rc *envoy_route.RouteConfiguration,
+	rules outbound.ResourceRules,
+	reader kri.ResourceReader,
+) error {
+	for _, vh := range rc.VirtualHosts {
+		for _, route := range vh.Routes {
+			if !strings.HasPrefix(route.Name, "kri_") {
```

Comment:
Should we have some common const so if we need to reference it we won't need to define each time/update each place?

Reply:
Added `kri.IsValid(s string) bool` to check if string is KRI